live/client.py

```diff
@@ -132,3 +147,52 @@ def stop_event(self, event_id):
         # Send request and do exception handling
         self.send_request(http_method="POST", url=url,
                           headers=headers, body=body)
+
+    def get_input_devices(self):
+        events_url = f'{self.server_ip}/live_events'
+        events_headers = self.generate_headers(events_url)
+        events_xml = self.send_request(
+            http_method="GET", url=events_url, headers=events_headers)
+
+        devices_url = f'{self.server_ip}/devices'
+        devices_headers = self.generate_headers(devices_url)
+        devices_xml = self.send_request(
+            http_method="GET", url=devices_url, headers=devices_headers)
+
+        events_list = ET.fromstring(events_xml.text)
+        devices_list = ET.fromstring(devices_xml.text)
+
+        # Find all devices info
+        devices_info = []
+        all_devices = set()
+        for device in devices_list:
+            all_devices.add(device.find('device_name').text)
+            device_dic = etree_to_dict(device)
+            devices_info.append(device_dic['device'])
+
+        # Find in use devices from all events
+        in_use_devices = set()
+        for event in events_list:
+            status = event.find('status').text
+            for input in event.findall('input'):
+                device_input = input.find('device_input')
+                if device_input:
+                    device_name = device_input.find('device_name')
+                    if status == 'preprocessing' or status == 'running':
+                        in_use_devices.add(device_name.text)
+
+        devices_availability = {}
+
+        for d in all_devices:
+            devices_availability[d] = True
+        for d in in_use_devices:
+            devices_availability[d] = False
+
+        # Append availability info to device info
+        for device in devices_info:
+            device['availability'] = \
+                devices_availability[device['device_name']]
+
+        devices_info = sorted(devices_info, key=lambda a: int(a["id"]))
+        print(json.dumps(devices_info))
+        return json.dumps(devices_info)
```

Comment:
The client method should the python object, the list of dicts.

In the controller, if we are using that in a http response, then we serialize to json, but not at this level.